src/main/java/com/kuljava/swiatwsi/services/RawMaterialsService.java

```diff
@@ -0,0 +1,32 @@
+package com.kuljava.swiatwsi.services;
+
+import com.kuljava.swiatwsi.rawmaterials.RawMaterials;
+import com.kuljava.swiatwsi.rawmaterials.RawMaterialsRepository;
+import com.kuljava.swiatwsi.world.Village;
+import com.kuljava.swiatwsi.world.VillageRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class RawMaterialsService {
+  @Autowired VillageRepository villageRepository;
+  @Autowired RawMaterialsRepository rawMaterialsRepository;
+
+  public Optional<RawMaterials> getRawMaterialsByVillageId(Long id) {
+    Optional<Village> village = villageRepository.findById(id);
+    return village.isEmpty()
```

Comment:
villageRepository.findById(id).).map(Village::GetRawMaterials).orElse(Optional::empty);

Reply:
It does't work, because rawMaterials is not functional interface